File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 */
public interface BrandService {
    /**
     * 查询所有
     *
     * @return
     */
    List<TbBrand> findAll();

    /**
     * 分页查询品牌
     *
     * @param pageNum  当前页
     * @param pageSize 每页记录
     * @return
     */
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 添加品牌
     *
     * @param tbBrand 品牌
     */
    void add(TbBrand tbBrand);

    /**
     * 根据id查询品牌
     *
     * @param id
     * @return
     */
    TbBrand findOne(Long id);

    /**
     * 修改品牌
     *
     * @param tbBrand
     */
    void update(TbBrand tbBrand);

    /**
     * 删除品牌
     *
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 分页模糊查询
     *
     * @param brand
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPage(TbBrand brand, Integer pageNum, Integer pageSize);

    /**
     * 返回下拉列表数据
     * @return
     */
    List<Map> selectOptionList();
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//品牌服务
app.service("brandService",function ($http) {
    this.findAll=function () {
        return $http.post("../brand/findAll.do");
    };

    this.findPage=function (page,rows) {
        return $http.get("../brand/findPage.do?page="+page+"&rows=" + rows);
    };


    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    };

    this.save=function (entity) {
        return $http.post("../brand/save.do",entity);
    };

    this.dele=function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    };


    this.search=function (page,rows,entity) {
        return $http.post("../brand/search.do?page="+page+"&rows="+rows,entity);
    }

    //下拉列表数据
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do");
    }
});